src/pybind/mgr/dashboard_v2/controllers/rgw.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import json
+
+from ..tools import ApiController, RESTController
+from .. import logger
+
+
+@ApiController('rgw')
+class Rgw(RESTController):
+
+    def __init__(self):
+        self.daemon = RgwDaemon()
+
+
+@ApiController()
+class RgwDaemon(RESTController):
+
+    def list(self):
+        daemons = []
+        for server in self.mgr.list_servers():
+            for service in server['services']:
+                if service['type'] == 'rgw':
+                    metadata = self.mgr.get_metadata('rgw', service['id'])
+                    status = self.mgr.get_daemon_status('rgw', service['id'])
+                    if 'json' in status:
+                        try:
+                            status = json.loads(status['json'])
+                        except ValueError:
+                            msg = "{0} had invalid status json"
+                            logger.warning(msg.format(service['id']))
+                            status = {}
+                    else:
+                        msg = '{0} has no key "json" in status'
+                        logger.warning(msg.format(service['id']))
+
+                    # extract per-daemon service data and health
+                    daemon = {
+                        'id': service['id'],
+                        'version': metadata['ceph_version'],
+                        'server_hostname': server['hostname'],
+                        'service': service,
+                        'server': server,
+                        'metadata': metadata,
+                        'status': status,
+                        'url': "{0}/api/rgw/daemon/{1}".format(
+                            self.mgr.url_prefix, service['id'])
+                    }
+
+                    daemons.append(daemon)
+
+        return {'daemons': sorted(daemons, key=lambda k: k['id'])}
+
+    def get(self, service_id):
```

Comment:
s/service_id/svc_id/ according to https://github.com/openattic/ceph/blob/wip-mgr-dashboard_v2/src/pybind/mgr/mgr_module.py#L395

Reply:
If possible, I'd like to keep `service_id`. It may be just my opinion but I like the expressiveness. Also, I wasn't able to translate `svc_id` to `service_id` when I first read it. And it's in the Zen of Python: Explicit is better than implicit. Of course there are different opinions about this and if yours or others are strong ones, let me know why that is.

Reply:
What's the name that the upstream Ceph project uses here? As we just discussed today, we should not change names on a whim, as it makes it impossible to find all occurrences of a variable across the entire code base.

Reply:
IMO I think in this particular case I would keep it as `service_id` since it's more descriptive.

If we were overriding some function of the `MgrModule` or other Ceph module then we should maintain the same signature and thus keep the same parameter names.

Reply:
Also, we are not changing the meaning of the parameter, we are just using a bit longer form of the same word

Reply:
I'd use `svc_id`  here, as the term `service_id` is not really used in ceph. I agree with Lenz here.